eva-accession-release/src/test/java/uk/ac/ebi/eva/accession/release/io/VariantContextWriterTest.java

```diff
@@ -0,0 +1,254 @@
+        File output = temporaryFolder.newFile();
+        assertWriteVcf(output, buildVariant(CHR_1, 1000, "C", "A", SNP_SEQUENCE_ONTOLOGY, STUDY_1, STUDY_2));
+
+        List<String> metadataLines = grepFile(output, STUDY_1);
+        assertEquals(1, metadataLines.size());
+        String[] infoPairs = metadataLines.get(0).split("\t")[7].split(";");
+        boolean isSidPresent = false;
+        for (String infoPair : infoPairs) {
+            String[] keyValue = infoPair.split("=");
+            if (keyValue[0].equals("SID")) {
+                isSidPresent = true;
+                String[] studies = keyValue[1].split(",");
+                assertEquals(2, studies.length);
+                assertEquals(Sets.newLinkedHashSet(STUDY_1, STUDY_2), Sets.newLinkedHashSet(studies));
+            }
+        }
+        assertTrue(isSidPresent);
+    }
+
+    @Test
+    public void checkSnpSequenceOntology() throws Exception {
+        checkSequenceOntology(SNP_SEQUENCE_ONTOLOGY, "C", "A");
+    }
+
+    private void checkSequenceOntology(String sequenceOntology, String reference, String alternate) throws Exception {
+        File output = temporaryFolder.newFile();
+        assertWriteVcf(output, buildVariant(CHR_1, 1000, reference, alternate, sequenceOntology, STUDY_1, STUDY_2));
+
+        List<String> metadataLines = grepFile(output, VARIANT_CLASS_KEY + "=");
+        assertEquals(1, metadataLines.size());
+        String[] infoPairs = metadataLines.get(0).split("\t")[7].split(";");
+        boolean isVariantClassPresent = false;
+        for (String infoPair : infoPairs) {
+            String[] keyValue = infoPair.split("=");
+            if (keyValue[0].equals(VARIANT_CLASS_KEY)) {
+                isVariantClassPresent = true;
+                String[] variantClass = keyValue[1].split(",");
+                assertEquals(1, variantClass.length);
+                assertEquals(sequenceOntology, variantClass[0]);
+            }
+        }
+        assertTrue(isVariantClassPresent);
+    }
+
+    @Test
+    public void checkInsertionSequenceOntology() throws Exception {
+        checkSequenceOntology(INSERTION_SEQUENCE_ONTOLOGY, "C", "CA");
+    }
+
+    @Test
+    public void checkDeletionSequenceOntology() throws Exception {
+        checkSequenceOntology(DELETION_SEQUENCE_ONTOLOGY, "CA", "A");
+    }
+
+    @Test
+    public void checkIndelSequenceOntology() throws Exception {
+        checkSequenceOntology(INDEL_SEQUENCE_ONTOLOGY, "CA", "A");
+    }
+
+    @Test
+    public void checkTandemRepeatSequenceOntology() throws Exception {
+        checkSequenceOntology(TANDEM_REPEAT_SEQUENCE_ONTOLOGY, "CA", "A");
+    }
+
+    @Test
+    public void checkSequenceAlterationSequenceOntology() throws Exception {
+        checkSequenceOntology(SEQUENCE_ALTERATION_SEQUENCE_ONTOLOGY, "CA", "A");
+    }
+
+    @Test
+    public void checkMnvSequenceOntology() throws Exception {
+        checkSequenceOntology(MNV_SEQUENCE_ONTOLOGY, "CA", "A");
+    }
+
```

Comment:
I get that all the above are make-believe variant constructions but could we make the refs and alt consistent with the SO term for clarity.